src/tests/index_btree_benchmark.cpp

```diff
@@ -0,0 +1,262 @@
+};
+
+struct VarValueSizeBtree {
+    using BtreeType = IndexTable< TestVarLenKey, TestVarLenValue >;
+    using KeyType = TestVarLenKey;
+    using ValueType = TestVarLenValue;
+    static constexpr btree_node_type leaf_node_type = btree_node_type::VAR_OBJECT;
+    static constexpr btree_node_type interior_node_type = btree_node_type::VAR_OBJECT;
+};
+
+struct VarObjSizeBtree {
+    using BtreeType = IndexTable< TestVarLenKey, TestVarLenValue >;
+    using KeyType = TestVarLenKey;
+    using ValueType = TestVarLenValue;
+    static constexpr btree_node_type leaf_node_type = btree_node_type::VAR_OBJECT;
+    static constexpr btree_node_type interior_node_type = btree_node_type::VAR_OBJECT;
+};
+
+struct PrefixIntervalBtree {
+    using BtreeType = IndexTable< TestIntervalKey, TestIntervalValue >;
+    using KeyType = TestIntervalKey;
+    using ValueType = TestIntervalValue;
+    static constexpr btree_node_type leaf_node_type = btree_node_type::PREFIX;
+    static constexpr btree_node_type interior_node_type = btree_node_type::FIXED;
+};
+
+template < typename TestType >
+struct IndexBtreeBenchmark : public BtreeHelper< TestType > {
+    using T = TestType;
+    using K = typename TestType::KeyType;
+    using V = typename TestType::ValueType;
+    class TestIndexServiceCallbacks : public IndexServiceCallbacks {
+    public:
+        TestIndexServiceCallbacks(IndexBtreeBenchmark* test) : m_benchmark(test) {}
+        std::shared_ptr< IndexTableBase > on_index_table_found(const superblk< index_table_sb >& sb) {
+            LOGINFO("Index table recovered");
+            LOGINFO("Root bnode_id {} version {}", sb->root_node, sb->link_version);
+            m_benchmark->m_bt = std::make_shared< typename T::BtreeType >(sb, m_benchmark->m_cfg);
+            return m_benchmark->m_bt;
+        }
+
+    private:
+        IndexBtreeBenchmark* m_benchmark;
+    };
+
+    IndexBtreeBenchmark() {
+        this->m_is_multi_threaded = true;
+        SetUp();
+    }
+
+    ~IndexBtreeBenchmark() { TearDown(); }
+
+    void SetUp() {
+        test_common::HSTestHelper::start_homestore(
+            "index_btree_benchmark",
+            {{HS_SERVICE::META, {.size_pct = 10.0}},
+             {HS_SERVICE::INDEX, {.size_pct = 70.0, .index_svc_cbs = new TestIndexServiceCallbacks(this)}}});
+
+        this->m_cfg = BtreeConfig(hs()->index_service().node_size());
+
+        auto uuid = boost::uuids::random_generator()();
+        auto parent_uuid = boost::uuids::random_generator()();
+
+        BtreeHelper< TestType >::SetUp();
+        this->m_bt = std::make_shared< typename T::BtreeType >(uuid, parent_uuid, 0, this->m_cfg);
+        hs()->index_service().add_index_table(this->m_bt);
+    }
+
+    void TearDown() {
+        BtreeHelper< TestType >::TearDown();
+        test_common::HSTestHelper::shutdown_homestore();
+    }
+
+    void ConcurrentOps(std::vector< std::string > input_ops) {
```

Comment:
NIT: This part is needed to be defined in helper since here, membtree and index use it. I will do it after your PR is merged